lib/plugins/azurepush.js

```diff
@@ -0,0 +1,323 @@
+          break;
+        case 'Android':
+          installation.platform = 'gcm';
+          installation.templates.nightscout.body = '{\"data\":{\"message\":\"$(message)\",\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"sound\":\"$(sound)\",\"title\":\"$(title)\"}}';
+          break;
+        case 'WinPhone': //Not yet supported
+        case 'Windows': //Not yet supported
+        default: //No platform specified, or invalid platform
+          // *****REMOVE*****: Beta logging
+          logTrelloIssue(env, '0x10-Register received without supported platform');
+          // END: *****REMOVE*****
+
+          return callback({message: 'Platform not supported'});
+      }
+
+      notificationHubService.createOrUpdateInstallation(installation, function (error, response) {
+        // *****REMOVE*****: Beta logging
+        if (error) {
+          logTrelloIssue(env, '0x11-Failed to create or update installation');
+        }
+        // END: *****REMOVE*****
+
+        callback(error, {installationId: installation.installationId});
+      });
+    }
+    else {
+      // *****REMOVE*****: Beta logging
+      logTrelloIssue(env, '0x12-Register received without platform');
+      // END: *****REMOVE*****
+
+      // No platform, required parameter
+      return callback({message: 'Platform required'});
+    }
+  };
+
+  azurepush.ack = function ack(request, callback) {
+    if (request && request.level) {
+      var response = {};
+      response.level = Number(request.level);
+      response.group = request.group || 'default';
+      response.time = request.time && Number(request.time);
+
+      //Convert minutes to milliseconds
+      if (response.time) {
+        response.time = response.time * 60 * 1000;
+      }
+
+      ctx.notifications.ack(response.level, response.group, response.time, true);
+
+      callback(null, response);
+    }
+    else {
+      // *****REMOVE*****: Beta logging
+      logTrelloIssue(env, '0x13-Ack received without level');
+      // END: *****REMOVE*****
+
+      //No level, required parameter
+      callback({message: 'Level required'});
+    }
+  };
+
+  if (notificationHubService) {
+    return azurepush;
+  }
+  else {
+    return null;
+  }
+}
+
+function setupAzurePush (env) {
+  if (env.settings.isEnabled('azurepush')) {
+    if (env.settings.baseURL) {
+      //TODO: Allow override
+      //env.extendedSettings.azurepush.hubName, env.extendedSettings.azurepush.connectionString
+      var notificationHubService = azure.createNotificationHubService('Push-Notification-Hub','Endpoint=sb://nspush.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=VdtFDvldTypVUgroq6JtrbI162upFh7MeWcpSdJu3yE=');
```

Comment:
If others have ideas, I'm open to them.  I struggled thinking of a way to avoid having the credentials available publicly (either on a setup guide or in the code) and placing an undue burden on getting the app connected up for push notifications.

I like the idea of a separate "credentialing" service, I'm just not sure how we would implement it without having to maintain **another** centralized service (that is prone to up-time guarantees and maintenance).

Let me know if you have other ideas!